crates/smdk/src/generate.rs

```diff
@@ -1,14 +1,51 @@
+use anyhow::{Error, Result};
 use clap::Parser;
-use anyhow::Result;
+use cargo_generate::{GenerateArgs, TemplatePath, generate};
+
+const GIT_TEMPLATE: &str = "https://github.com/infinyon/fluvio-smartmodule-template";
 
 /// Generate new SmartModule project
 #[derive(Debug, Parser)]
 pub struct GenerateOpt {
     name: String,
 }
+
 impl GenerateOpt {
     pub(crate) fn process(&self) -> Result<()> {
         println!("Generating new SmartModule project: {}", self.name);
+        let template_path = TemplatePath {
+            git: Some(String::from(GIT_TEMPLATE)),
+            auto_path: None,
+            subfolder: None,
+            test: false,
+            branch: None,
+            tag: None,
```

Comment:
I think we could keep different versions of the SM templates in https://github.com/infinyon/fluvio-smartmodule-template and use this `tag` field to specify which to use. WDYT?

Reply:
No opinion.  There are so many ways to slice templates.  Key factor is reduce duplication so no repeating logic